src/modules/drawing/lib/DrawingManager.js

```diff
@@ -0,0 +1,242 @@
+// FIXME: change cursor
+// FIXME: add tooltips
+// FIXME: apply new style function?
+// FIXME: linestring or polygon
+// FIXME: use feature properties for styling
+// FIXME: no zoom on double click
+// FIXME: right click to remove point while drawing ?
+
+import { noModifierKeys, singleClick } from 'ol/events/condition'
+import Event from 'ol/events/Event'
+import GeoJSON from 'ol/format/GeoJSON'
+import KML from 'ol/format/KML'
+import DrawInteraction from 'ol/interaction/Draw'
+import ModifyInteraction from 'ol/interaction/Modify'
+import SelectInteraction from 'ol/interaction/Select'
+import SnapInteraction from 'ol/interaction/Snap'
+import VectorLayer from 'ol/layer/Vector'
+import Observable from 'ol/Observable'
+import VectorSource from 'ol/source/Vector'
+import { getUid } from 'ol/util'
+import { featureStyle } from './style'
+
+class ChangeEvent extends Event {
+    /**
+     * @param {Object} geojson Features in GeoJSON format
+     * @param {string} kml Features in KML format
+     * @param {string} [featureId] Feature id
+     */
+    constructor(geojson, kml, featureId) {
+        super('change')
+
+        this.geojson = geojson
+
+        this.kml = kml
+
+        this.featureId = featureId
+    }
+}
+
+class SelectEvent extends Event {
+    /**
+     * @param {string | null} featureId Feature id
+     * @param {string | null} featureType Feature type
+     * @param {number[]} coordinates Pointer coordinates
+     * @param {boolean} modifying
+     */
+    constructor(featureId, featureType, coordinates, modifying) {
+        super('select')
+
+        this.featureId = featureId
+
+        this.featureType = featureType
+
+        this.coordinates = coordinates
+
+        this.modifying = modifying
+    }
+}
+
+export default class DrawingManager extends Observable {
```

Comment:
Some brief explanation of what this manager does, and its main function/methods, would be great